Source/IntegrationTests/TestFiles/LitTests/LitTest/comp/placeholders/placeholderHappyflow.dfy

```diff
@@ -0,0 +1,67 @@
+// RUN: %testDafnyForEachCompiler "%s"
+
+placeholder module FooModule {
+  method Foo() returns (i: int) 
+    ensures i >= 2
+}
+
+abstract module SuchAbstract {
+  function OrNot(): int {
+    1
+  }
+}
+
+placeholder module BarModule refines SuchAbstract {
+  method Bar() returns (i: int) 
+    ensures i >= 1
+}
+
+
+placeholder module BazModule replaces BarModule {
```

Comment:
This is nice, the fact that you can have chains of replacement means I don't personally care much that you can't refine after you placehold.